### ISSUE TITLE
Yolov8 trainning freezed by unknown issue

### ISSUE THREAD

**Issue by ChrisChenSar1eL** [ultralytics/ultralytics#20415]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

YOLOv8 trainning will freeze at random epoches, images of loading freezed and the volatile of gpu went down 0%. 
there isn't any error or warning in command line. and i found this problem in trainning both yolov8 and yolov11. 
![Image](https://github.com/user-attachments/assets/1ed6f757-fa2b-48e2-80a0-5de0d6f6823d)

### Environment

Ultralytics 8.3.116 🚀 Python-3.8.11 torch-1.10.1+cu111 CUDA:0 (NVIDIA GeForce RTX 3090, 24226MiB)
Setup complete ✅ (32 CPUs, 125.6 GB RAM, 1140.8/3666.5 GB disk)

OS                  Linux-6.8.0-57-generic-x86_64-with-glibc2.27
Environment         Linux
Python              3.8.11
Install             git
Path                /home/ultralytics/ultralytics
RAM                 125.55 GB
Disk                1140.8/3666.5 GB
CPU                 13th Gen Intel Core(TM) i9-13900K
CPU count           32
GPU                 NVIDIA GeForce RTX 3090, 24226MiB
GPU count           2
CUDA                11.1

numpy               ✅ 1.23.2<=2.1.1,>=1.23.0
matplotlib          ✅ 3.7.4>=3.3.0
opencv-python       ✅ 4.8.1.78>=4.6.0
pillow              ✅ 10.1.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.31.0>=2.23.0
scipy               ✅ 1.10.1>=1.4.1
torch               ✅ 1.10.1+cu111>=1.8.0
torchvision         ✅ 0.11.2+cu111>=0.9.0
tqdm                ✅ 4.66.1>=4.64.0
psutil              ✅ 5.9.7
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.0.3>=1.1.4
seaborn             ✅ 0.13.0>=0.11.0
ultralytics-thop    ✅ 2.0.14>=2.0.0

### Minimal Reproducible Example

yolo task=segment mode=train model=yolov8n-seg.pt \
        data=./datasets/golfClub.yaml \
        epochs=100 \
        imgsz=640 \
        batch=16 \
        device=1 \
        project='./segment_exps' \
        #pretrained=True\
        #workers=0 \
        cache=True

### Additional

i have met this problem on two different pc, both rtx 3090. i just want to know the reason and how can i avoid this problem.
here is another pc env:
Ultralytics 8.3.84 🚀 Python-3.10.12 torch-2.5.1+cu124 CUDA:0 (NVIDIA GeForce RTX 3090, 24576MiB)
Setup complete ✅ (16 CPUs, 15.5 GB RAM, 49.0/250.9 GB disk)

OS                  Linux-5.15.133.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
Environment         Linux
Python              3.10.12
Install             pip
Path                /home/sare/.local/lib/python3.10/site-packages/ultralytics
RAM                 15.55 GB
Disk                49.0/250.9 GB
CPU                 Intel Core(TM) i7-10700K 3.80GHz
CPU count           16
GPU                 NVIDIA GeForce RTX 3090, 24576MiB
GPU count           1
CUDA                12.4

numpy               ✅ 1.25.2<=2.1.1,>=1.23.0
matplotlib          ✅ 3.8.3>=3.3.0
opencv-python       ✅ 4.6.0.66>=4.6.0
pillow              ✅ 10.2.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.28.2>=2.23.0
scipy               ✅ 1.9.3>=1.4.1
torch               ✅ 2.5.1+cu124>=1.8.0
torch               ✅ 2.5.1+cu124!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.20.1+cu124>=0.9.0
tqdm                ✅ 4.65.2>=4.64.0
psutil              ✅ 5.9.8
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.1>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.13>=2.0.0

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @ChrisChenSar1eL, thank you for your detailed report and for your interest in Ultralytics 🚀! This is an automated response to help you get support quickly—a member of the Ultralytics engineering team will assist you further soon.

We recommend visiting our [Docs](https://docs.ultralytics.com/) for guidance, where you’ll find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples, troubleshooting tips, and answers to common questions.

Since this is a 🐛 Bug Report, thank you for providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) and detailed environment information—it’s very helpful for debugging!

If you have any other relevant logs, dataset samples, or details about when the freeze occurs (such as specific epochs, use of `cache`, or worker settings), please add them here to help us investigate further.

Join the Ultralytics community for real-time chat on [Discord](https://discord.com/invite/ultralytics) 🎧, in-depth discussions on [Discourse](https://community.ultralytics.com/), or share knowledge on our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Please ensure you are using the latest `ultralytics` package and dependencies. Sometimes, issues like this are resolved in newer releases. Upgrade in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) as follows:

```bash
pip install -U ultralytics
```
See all requirements in [pyproject.toml](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml).

## Environments

You can run YOLO in any of these up-to-date, verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you again for your contribution! An engineer will assist you as soon as possible.

**Comment by Y-T-G** [comment]
What if you don't use `cache=True`?
